imputation_functions.py

```diff
@@ -51,61 +79,58 @@ def factors_calculation_a(row, questions, parameters):
 
     :return: row of DataFrame
     """
-    # extract the parameters:
-    first_threshold = parameters["first_threshold"]
-    second_threshold = parameters["second_threshold"]
-    third_threshold = parameters["third_threshold"]
-    first_imputation_factor = parameters["first_imputation_factor"]
-    second_imputation_factor = parameters["second_imputation_factor"]
-    third_imputation_factors = parameters["third_imputation_factors"]
-    region_column = parameters["region_column"]
-    regionless_code = parameters["regionless_code"]
-    survey_column = parameters["survey_column"]
-    percentage_movement = parameters["percentage_movement"]
-    distinct_values = parameters["distinct_values"]
+
+    runtime_variables, errors = ExtendedFactorsCalculationASchema().load(kwargs)
+    if errors:
+        raise ValueError(f"Error validating factors params: {errors}")
+
+    rv = SimpleNamespace(**runtime_variables)
+
     for question in questions:
-        if row[region_column] == regionless_code:
-            if row[survey_column] == "066":
-                if row["movement_" + question + "_count"] < int(first_threshold):
-                    row["imputation_factor_" + question] = float(first_imputation_factor)
+        if row[rv.region_column] == rv.regionless_code:
+            if row[rv.survey_column] == "066":
+                if row["movement_" + question + "_count"] < int(rv.first_threshold):
+                    row["imputation_factor_" + question] =\
+                        float(rv.first_imputation_factor)
                 else:
                     row["imputation_factor_" + question] =\
                         float(pd.to_numeric(row["mean_" + question]))
-            elif row[survey_column] == "076":
-                if row["movement_" + question + "_count"] < int(second_threshold):
-                    row["imputation_factor_" + question] = float(second_imputation_factor)
+            elif row[rv.survey_column] == "076":
```

Comment:
Do we have a TD somewhere to remove the magic values?

Reply:
You mean the 066 and 076? I believe we said because all three surveys do this section differently we wouldn't bother with the hassle so there won't be any tech debt related to it.